### PR TITLE
Implement Padding property in ImageButtonHandler

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Implement **Padding** property in ImageButton.

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)

#### Does this PR touch anything that might affect accessibility?
- No

### CODE REVIEW

**Review by rmarinho — APPROVED** [review]

File: src/Core/src/Platform/Android/ImageButtonExtensions.cs [hunk 1]
```diff
@@ -29,5 +29,22 @@ public static void UpdateCornerRadius(this ShapeableImageView nativeButton, IBut
...
+			{
+				return;
+			}
+
+			nativeButton.SetContentPadding(		
+				(int)context.ToPixels(imageButton.Padding.Left),
+				(int)context.ToPixels(imageButton.Padding.Top),
+				(int)context.ToPixels(imageButton.Padding.Right),
+				(int)context.ToPixels(imageButton.Padding.Bottom)
+				);
```
Comment (mattleibow) [inline src/Core/src/Platform/Android/ImageButtonExtensions.cs:47]: How does this work with default paddings and 0 or -1? I know there is a bunch of logic with the "default"/"normal" button. Most notably, we need to make sure the default button looks correct and a padding of 0 does not have a negative space and is in fact a 0 padding.

**Comment by Redth** [comment]
@jsuarezruiz  I don't see the padding value influencing MacCatalyst at all in the ImageButton gallery page.

**Comment by J-Swift** [comment]
@jsuarezruiz I happened to be looking into this because I noticed it wasn't implemented yet, and noticed that your Android implementation won't work as expected due to an Android Framework bug which pads the entire button rather than just the content area (https://github.com/material-components/material-components-android/issues/2063).

This is what that looks like:
Screen Shot 2022-03-03 at 10 11 32 PM

I propose the following for the Android implementation:

```
public static void UpdatePadding(this ShapeableImageView platformButton, IImageButton imageButton)
{
		var pad = imageButton.Padding;

		// NOTE(jpr): post on handler to get around an Android Framework bug.
		// see: https://github.com/material-components/material-components-android/issues/2063
		platformButton.Post(() =>
		{
				platformButton.SetContentPadding(
					(int)platformButton.Context.ToPixels(pad.Left),
					(int)platformButton.Context.ToPixels(pad.Top),
					(int)platformButton.Context.ToPixels(pad.Right),
					(int)platformButton.Context.ToPixels(pad.Bottom)
				);
		});
}
```

which results in this

Screen Shot 2022-03-03 at 10 13 53 PM

**Review by mattleibow — CHANGES_REQUESTED** [review]
The padding may be wrong and there might be an android bug that we need to work around.

**Comment by jsuarezruiz** [comment]
> @jsuarezruiz I happened to be looking into this because I noticed it wasn't implemented yet, and noticed that your Android implementation won't work as expected due to an Android Framework bug which pads the entire button rather than just the content area ([material-components/material-components-android#2063](https://github.com/material-components/material-components-android/issues/2063)).
> 
> This is what that looks like: Screen Shot 2022-03-03 at 10 11 32 PM
> 
> I propose the following for the Android implementation:
> 
> ```
> public static void UpdatePadding(this ShapeableImageView platformButton, IImageButton imageButton)
> {
> 		var pad = imageButton.Padding;
> 
> 		// NOTE(jpr): post on handler to get around an Android Framework bug.
> 		// see: https://github.com/material-components/material-components-android/issues/2063
> 		platformButton.Post(() =>
> 		{
> 				platformButton.SetContentPadding(
> 					(int)platformButton.Context.ToPixels(pad.Left),
> 					(int)platformButton.Context.ToPixels(pad.Top),
> 					(int)platformButton.Context.ToPixels(pad.Right),
> 					(int)platformButton.Context.ToPixels(pad.Bottom)
> 				);
> 		});
> }
> ```
> 
> which results in this
> 
> Screen Shot 2022-03-03 at 10 13 53 PM

Used the Jimmy workaround. Thanks for include the issue.

**Review by rmarinho — CHANGES_REQUESTED** [review]
Tests are failing on Android

https://dev.azure.com/xamarin/public/_build/results?buildId=58437&view=ms.vss-test-web.build-test-results-tab

**Pull request merged**